osclib/stagingapi.py

```diff
@@ -998,16 +998,17 @@ def map_ring_package_to_subject(self, project, pkg):
         """
         # it's actually a pretty stupid algorithm, but it might become more complex later
 
-        if project.endswith(':DVD'):
-            return project  # not yet
+        # assuming it is in adi staging, workaround for https://progress.opensuse.org/issues/9646
+        if self.is_adi_project(project) and self.item_exists(project, pkg):
```

Comment:
not just submit_to_prj(), those code also introduced in rm_from_prj(), the code is mainly to protect a special case like https://progress.opensuse.org/issues/9646 mentioned, if pkg A already staged in adi staging and meanwhile pkg A was added to ring2, while adi accepting it got trouble to delete the package from project as tool thinks it's ring2 package must cleanup package from :DVD, however doesn't have it in adi staging.